### DISCUSSION TITLE
Menu position when Menubar overflow-x

### DISCUSSION THREAD

**Discussion by lishine** [ariakit/ariakit#782] [Questions] [answered]
When the Menubar or it's closest horizonatal container is put in overflow-x and position=relative (or the container of button+menu is in position=relative), then the menu is not behaving like absolute positioned.
Like in the picture.
![image](https://user-images.githubusercontent.com/6741645/98100629-ce9a9980-1e99-11eb-97b2-0cd0563b02c9.png)

One solution is putting position=relative on the whole panel which holds this horizontal row of filters. But then when scrolling right, the menu also move right. And does not help setting     ``` const menu = useMenuState({ unstable_flip: true, unstable_preventOverflow: true })```
Like in the second picture.
![image](https://user-images.githubusercontent.com/6741645/98100573-b9be0600-1e99-11eb-959f-6275789fb978.png)

**Comment by lishine** [comment]
Also, in the second picture, you see that the menu is flowing out of the white panel, while it should be placed inside

**Comment by diegohaz** [comment]
Can you provide a CodeSandbox reproduction of that? I think it's solvable with Popper.js, but I would like to play with the code a little bit.

**Reply by lishine** [reply]
Here you go
https://codesandbox.io/s/reakit-menu-overflow-x-f6o11
There the position=relative in two places:
Causing either drop down menu to be covered
or the  menu not moving left/right according to placement

**Comment by prbxr** [comment]
@diegohaz Do you manage to do something?
If not, maybe give me direction of investigation?

**Comment by diegohaz** [comment, marked as answer]
You can render `<Menu>` inside `<Portal>` and this solves the overflow problem in the first picture:

```jsx
import { Portal, Menu } from "reakit";

<Portal>
  <Menu />
</Portal>
```

But there's currently no solution for the second picture as far as I know. Could you please create an issue as a feature request for that?
